core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java

```diff
@@ -283,14 +284,21 @@ public boolean removeKey(ECKey key) {
         }
     }
 
+    /**
+     * Returns the earliest creation time of keys in this chain.
+     * @return earliest creation times of keys in this chain,
+     *         {@link Instant#EPOCH} if at least one time is unknown,
+     *         {@link Instant#MAX} if no keys in this chain
+     */
     @Override
-    public long getEarliestKeyCreationTime() {
+    public Instant getEarliestKeyCreationTimeInstant() {
         lock.lock();
         try {
             return hashToKeys.values().stream()
                     .mapToLong(ECKey::getCreationTimeSeconds)
-                    .min()
-                    .orElse(Long.MAX_VALUE);
+                    .mapToObj(t -> t == Long.MAX_VALUE ? Instant.MAX : Instant.ofEpochSecond(t))
```

Comment:
We should probably create a utility static method for this. Then a method reference could be used here.

Reply:
> We should probably create a utility static method for this.

I meant a static utility method for seconds-to-Instant like this:

public Instant secondsToInstantWithMax(long t) {
    return t == Long.MAX_VALUE ? Instant.MAX : Instant.ofEpochSecond(t);
 }
 
 There's probably a better name.

Reply:
and this is probably irrelevant because "the code will go away as soon as getCreationTime() will migrate to Instant"